src/UnityMvvmToolkit.UnityPackage/Assets/Plugins/UnityMvvmToolkit/Runtime/UITK/BindableUIElements/BindableDropdownField.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Runtime.CompilerServices;
+using UnityEngine.UIElements;
+using UnityMvvmToolkit.Common.Interfaces;
+using UnityMvvmToolkit.Core;
+using UnityMvvmToolkit.Core.Extensions;
+using UnityMvvmToolkit.Core.Interfaces;
+
+namespace UnityMvvmToolkit.UITK.BindableUIElements
+{
+    public partial class BindableDropdownField : DropdownField, IBindableCollection, IInitializable, IDisposable
+    {
+        private IProperty<string> _selectedItemProperty;
+        private IReadOnlyProperty<ObservableCollection<string>> _itemsSource;
+
+        private PropertyBindingData _selectedItemBindingData;
+        private PropertyBindingData _itemsSourceBindingData;
+        
+        
+        public void Initialize()
+        {
+            choices = new List<string>();
+        }
+
+        public void Dispose()
+        {
+            choices.Clear();
+        }
+        
+        public void SetBindingContext(IBindingContext context, IObjectProvider objectProvider)
+        {
+            if (string.IsNullOrWhiteSpace(BindingItemsSourcePath) == false)
+            {
+                _itemsSourceBindingData ??= BindingItemsSourcePath.ToPropertyBindingData();
+                _itemsSource = objectProvider
+                    .RentReadOnlyProperty<ObservableCollection<string>>(context, _itemsSourceBindingData);
+                _itemsSource.Value.CollectionChanged += OnItemsCollectionChanged;
+                choices = new List<string>(_itemsSource.Value);
+            }
+            
+            if (string.IsNullOrWhiteSpace(BindingSelectedItemPath) == false)
+            {
+                _selectedItemBindingData ??= BindingSelectedItemPath.ToPropertyBindingData();
+                _selectedItemProperty = objectProvider.RentProperty<string>(context, _selectedItemBindingData);
+                _selectedItemProperty.ValueChanged += OnSelectedItemValueChanged;
+                UpdateControlValue(_selectedItemProperty.Value);
+                this.RegisterValueChangedCallback(OnControlValueChanged);
+                _selectedItemProperty.Value = choices[0];
```

Comment:
Possible `OutOfRangeException` when the `BindingItemsSourcePath` is not set or `_itemsSource.Value` is empty.

I missed this yesterday.

You have to check both `BindingItemsSourcePath` and `BindingSelectedItemPath` at the beginning.

```csharp
public void SetBindingContext(IBindingContext context, IObjectProvider objectProvider)
{
    if (string.IsNullOrWhiteSpace(BindingItemsSourcePath) ||
        string.IsNullOrWhiteSpace(BindingSelectedItemPath))
    {
        return;
    }
    
    ...
}
```

Reply:
I think we need to check the parameters separately. Therefore, I wrote a null check